### PR TITLE
GR-83, non-root systems can update others' attributes

### PR DESCRIPTION
**Pull request by knutsenm**
@twinge, note that an unrelated test unexpectedly broke. Also, I have questions about which HTTP error codes ought to be used. As a bonus, I removed systems_i_trust.

### CODE REVIEW

File: app/controllers/systems_controller.rb [hunk 1]
```diff
@@ -7,8 +7,12 @@ def index
   end
 
   def update
-    system.update_attributes!(system_params)
-    render_id(system)
+    if root_or_own?
+      system.update_attributes!(system_params)
+      render_id(system)
+    else
+      render_not_found # MK: render_forbidden?
```
Comment (knutsenm) [inline app/controllers/systems_controller.rb]: Suggest this should be render_forbidden (403) to match #update in other controllers.

File: app/controllers/systems_controller.rb [hunk 2]
```diff
@@ -16,7 +20,7 @@ def create
       @system = System.create!(system_params)
       render json: system, scope: { root: true }, status: :created
     else
-      render_not_found
+      render_not_found # MK: render_forbidden?
```
Comment (knutsenm) [inline app/controllers/systems_controller.rb]: Suggest this (and two more below) should be render_forbidden (403) to match #update in this controller.

File: spec/models/entity_spec.rb [hunk 3]
```diff
@@ -128,7 +128,7 @@
         entity.send(:_assign_attribute, 'name', %w(Bill Bob))
 
         json = EntitySerializer.new(entity).as_json
-        expect(json[:entity][:team][:name]).to eq %w(Bill Bob)
+        expect(json[:entity][:team][:name].sort).to eq %w(Bill Bob)
```
Comment (knutsenm) [inline spec/models/entity_spec.rb]: This unexpectedly began returning %w(Bob Bill)

File: db/migrate/20150206225310_remove_systems_i_trust_from_systems.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class RemoveSystemsITrustFromSystems < ActiveRecord::Migration
+  def change
+    remove_column :systems, :systems_i_trust
```
Comment (knutsenm) [inline db/migrate/20150206225310_remove_systems_i_trust_from_systems.rb]: Bonus removal of systems_i_trust, per comments in GR-83

File: app/controllers/systems_controller.rb (hunk 1 above)
Comment (twinge) [inline app/controllers/systems_controller.rb]: :+1:

File: app/controllers/systems_controller.rb (hunk 2 above)
Comment (twinge) [inline app/controllers/systems_controller.rb]: :+1:

File: db/migrate/20150206225310_remove_systems_i_trust_from_systems.rb (hunk 4 above)
Comment (twinge) [inline db/migrate/20150206225310_remove_systems_i_trust_from_systems.rb]: :+1:

**Comment by twinge** [comment]
:+1:

**Pull request merged**